src/core/effects/qgsimageoperation.cpp

```diff
@@ -717,6 +717,9 @@ void QgsImageOperation::GaussianBlurOperation::operator()( QgsImageOperation::Im
       destRef = reinterpret_cast< QRgb * >( outputLineRef );
       for ( int x = 0; x < width; ++x, ++destRef, sourceRef += 4 )
       {
+        if ( mFeedback && mFeedback->isCanceled() )
+          break;
```

Comment:
Should not be just return instead of break?

Reply:
I think break is safest, just in case clean up code at the end of the function is added later